gui/layers/_vectors_layer/view/properties.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+# title           : properties.py
+# description     :qt vectors layer
+# author          :bryant.chhun
+# date            :1/16/19
+# version         :0.0
+# usage           :
+# notes           :
+# python_version  :3.6
+
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import QLabel, QComboBox, QSpinBox
+
+from ..._base_layer import QtLayer
+
+
+class QtVectorsLayer(QtLayer):
+
+    def __init__(self, layer):
+        super().__init__(layer)
+
+        # vector color adjustment and widget
+        face_comboBox = QComboBox()
+        colors = self.layer._colors
+        for c in colors:
+            face_comboBox.addItem(c)
+        index = face_comboBox.findText(self.layer.color, Qt.MatchFixedString)
+        if index >= 0:
+            face_comboBox.setCurrentIndex(index)
+        face_comboBox.activated[str].connect(
+            lambda text=face_comboBox: self.changeFaceColor(text))
+        self.grid_layout.addWidget(QLabel('color:'), 3, 0)
+        self.grid_layout.addWidget(face_comboBox, 3, 1)
+
+        # line width in pixels
```

Comment:
Is there a reason you prefer the `QComboBox` to the `QSlider` in this case? Could have `QSlider` and a `QTextBox`?

Reply:
If you're referring to the "color" widget, i just copied the implementation from markers layer.
If you're referring to width, then yes I could try a slider.  Thing is that the documentation here says width is in pixels (http://api.vispy.org/en/latest/visuals.html#vispy.visuals.LineVisual).  Generally I like the precision of the numbers over slider.
Also, I did try slider for the length widget, but it would call too many updates.  And because length adjusts the data, the interaction would slow down dramatically.

Reply:
I was referring to the width. I am fine with the `QSpinBox`, maybe I will change it in the markers / shapes layers too for consistency. I take your point that it is nice to have that precision